src/use-swr-infinite.ts

```diff
@@ -29,58 +29,43 @@ type SWRInfiniteResponseInterface<Data = any, Error = any> = responseInterface<
 }
 
 function useSWRInfinite<Data = any, Error = any>(
-  getKey: KeyLoader<Data>
-): SWRInfiniteResponseInterface<Data, Error>
-function useSWRInfinite<Data = any, Error = any>(
-  getKey: KeyLoader<Data>,
-  config?: SWRInfiniteConfigInterface<Data, Error>
-): SWRInfiniteResponseInterface<Data, Error>
-function useSWRInfinite<Data = any, Error = any>(
-  getKey: KeyLoader<Data>,
-  fn?: fetcherFn<Data>,
-  config?: SWRInfiniteConfigInterface<Data, Error>
-): SWRInfiniteResponseInterface<Data, Error>
-function useSWRInfinite<Data = any, Error = any>(
-  ...args
+  ...args:
+    | readonly [KeyLoader<Data>]
+    | readonly [KeyLoader<Data>, fetcherFn<Data>]
+    | readonly [KeyLoader<Data>, SWRInfiniteConfigInterface<Data, Error>]
+    | readonly [
+        KeyLoader<Data>,
+        fetcherFn<Data>,
+        SWRInfiniteConfigInterface<Data, Error>
+      ]
 ): SWRInfiniteResponseInterface<Data, Error> {
-  let getKey: KeyLoader<Data>,
-    fn: fetcherFn<Data> | undefined,
-    config: SWRInfiniteConfigInterface<Data, Error> = {}
+  const getKey = args[0]
 
-  if (args.length >= 1) {
-    getKey = args[0]
-  }
-  if (args.length > 2) {
-    fn = args[1]
-    config = args[2]
-  } else {
-    if (typeof args[1] === 'function') {
-      fn = args[1]
-    } else if (typeof args[1] === 'object') {
-      config = args[1]
-    }
-  }
-
-  config = Object.assign(
+  const config = Object.assign(
     {},
     defaultConfig,
     useContext(SWRConfigContext),
-    config
+    args.length === 3
```

Comment:
Let's keep the exact same behavior:

```suggestion
    args.length > 2
```